libs/elodin-editor/src/plugins/camera_anchor.rs

```diff
@@ -0,0 +1,21 @@
+use bevy::math::DVec3;
+use bevy::transform::components::Transform;
+
+/// Returns the camera's view-space anchor when looking at the world origin.
+///
+/// This falls back to `None` if the transform cannot be inverted (for example,
+/// because it contains invalid scale values), which signals the caller to let
+/// `EditorCam` reuse its last known depth instead of corrupting it with NaNs.
+pub fn camera_anchor_from_transform(transform: &Transform) -> Option<DVec3> {
+    let anchor = transform
+        .compute_matrix()
+        .as_dmat4()
+        .inverse()
+        .transform_point3(DVec3::ZERO);
```

Comment:
For this `transform_point3`, you can read off the fourth column from the matrix. I double-checked against the implementation of [`transform_point3`](https://docs.rs/glam/0.30.8/src/glam/f64/dmat4.rs.html#1169), and it had no surprises. So instead of `transform_point`, you can put `w_axis.xyz()`.

I also confirmed via Mathematica.

<img width="550" height="406" alt="Screenshot 2025-11-24 at 2 48 45 PM" src="https://github.com/user-attachments/assets/8f2ed890-6c69-4f1f-bbf7-718f65b47017" />